metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/utilities/CSVUtilities.java

```diff
@@ -29,81 +17,41 @@ public class CSVUtilities {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CSVUtilities.class);
 
-    private final MongoMetisCoreDao mongoMetisCoreDao;
-    public CSVUtilities(MongoMetisCoreDao mongoMetisCoreDao) {
-        this.mongoMetisCoreDao = mongoMetisCoreDao;
+    private final PerformanceMetricsUtilities performanceMetricsUtilities;
+    public CSVUtilities(PerformanceMetricsUtilities performanceMetricsUtilities) {
+        this.performanceMetricsUtilities = performanceMetricsUtilities;
     }
 
-    public void writeIntoCsvFile(String filePath, Date startDate, Date endDate) throws FileNotFoundException {
+    public void writeMetric1IntoCsvFile(String filePath, LocalDateTime startDate, LocalDateTime endDate) {
```

Comment:
If you divide the metrics in classes, you can define an abstract class called (e.g.Metric) with an abstract method toCSV.
so when implementing you use for Metric1 class use what is in writeMetric1IntoCsvFile and for Metric2 class use what is in writeMetric2IntoCsvFile. So the code abstraction will improve in the sense that each metric will have their own logic in the right place. in the main class will look something like.
```
metric1.toCsv(...)

metric2.toCsv(...)
```